csrc/executor_utils.cpp

```diff
@@ -465,9 +536,17 @@ void validateAlignedVectorizedFusionInputOutput(
 
     // If this domain is contiguous or size == 1, then not necessary to check
     // the stride. Otherwise, stride must be 1 if it's rightmost or
-    // divisible by word_size
+    // divisible by word_size.
+
+    // Note that when a domain to the right of
+    // this domain is sliced, this domain is no longer contiguous.
+    bool non_contig_due_to_slice = i < (int64_t)sizes.size() - 1 &&
+        sliced_domains.count(
+            domain_to_validate.at(no_reduction_to_full.at(i + 1)));
```

Comment:
What if we have `[I1, b, slice]`?

Reply:
Would there be a problem? The broadcast domain would pass as its `size` should be 1. The sliced domain should also pass.

Reply:
I think I am more worried about missed fail. For example, if
```
t0 = [4, 1025, 3]
t1 = t0[:, :1024, 3]
```
For this case we should fail, because we might have misaligned address on `t1[1,...]`. And currently, we are correctly failing the `stride % word_size == 0` check. However, if I change my fusion into
```
t0 = [4, 1, 1025, 3]
t1 = t0[:, :, :1024, 3]
```
then we will not fail as expected, because dim 0 is no longer `non_contig_due_to_slice` any more, and we might treat dim 0 as contiguous and not checking its stride.

Reply:
That's a nice catch. Thanks! Updated the code and added the case as a test.